src/geometry/dual_quaternion.rs

```diff
@@ -0,0 +1,126 @@
+use crate::{Quaternion, RealField, Scalar, SimdRealField};
+use simba::simd::SimdValue;
+
+#[cfg(feature = "serde-serialize")]
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+
+/// A dual quaternion.
+///
+/// NOTE:
+///  As of December 2020, dual quaternion support is a work in progress.
+///  If a feature that you need is missing, feel free to open an issue or a PR.
+///  See https://github.com/dimforge/nalgebra/issues/487
+#[repr(C)]
+#[derive(Debug)]
+pub struct DualQuaternion<N: Scalar + SimdValue> {
+    /// The rotation part of the dual quaternion.
+    pub rot: Quaternion<N>,
+    /// The translation part of the dual quaternion.
+    pub trans: Quaternion<N>,
+}
+
+impl<N: RealField> Default for DualQuaternion<N> {
+    fn default() -> Self {
+        Self {
+            rot: Quaternion::default(),
+            trans: Quaternion::default(),
+        }
+    }
+}
+
+impl<N: SimdRealField + Eq> Eq for DualQuaternion<N> where N::Element: SimdRealField {}
+
+impl<N: SimdRealField> PartialEq for DualQuaternion<N>
+where
+    N::Element: SimdRealField,
+{
+    fn eq(&self, rhs: &Self) -> bool {
+        self.rot == rhs.rot && self.trans == rhs.trans
+    }
+}
+
+impl<N: Scalar + Copy + SimdValue> Copy for DualQuaternion<N> {}
+
+impl<N: Scalar + SimdValue> Clone for DualQuaternion<N> {
+    #[inline]
+    fn clone(&self) -> Self {
+        Self {
+            rot: self.rot.clone(),
+            trans: self.trans.clone(),
+        }
+    }
+}
+
+impl<N: SimdRealField + Scalar + SimdValue> DualQuaternion<N>
+where
+    N::Element: SimdRealField,
+{
+    /// Normalizes this quaternion.
+    ///
+    /// # Example
+    /// ```
+    /// # #[macro_use] extern crate approx;
+    /// # use nalgebra::{DualQuaternion,Quaternion};
+    /// let rot = Quaternion::new(1.0, 2.0, 3.0, 4.0);
+    /// let trans = Quaternion::new(5.0, 6.0, 7.0, 8.0);
+    ///
+    /// let dq = DualQuaternion::new(rot,trans);
+    ///
+    /// let dq_normalized = dq.normalize();
+    ///
+    /// relative_eq!(dq_normalized.rot.norm(), 1.0);
+    /// ```
+    #[inline]
+    #[must_use = "Did you mean to use normalize_mut()?"]
+    pub fn normalize(&self) -> Self {
```

Comment:
It would be great to add an implementation of `normalize_mut` too, since it is mentioned here.